tests/OmniSharp.MSBuild.Tests/OnDemandProjectsLoadTests.cs

```diff
@@ -0,0 +1,162 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using OmniSharp.Models;
+using OmniSharp.Models.MembersTree;
+using OmniSharp.Models.V2.CodeActions;
+using OmniSharp.Models.V2.CodeStructure;
+using OmniSharp.MSBuild.Models;
+using OmniSharp.Options;
+using OmniSharp.Roslyn.CSharp.Services.Refactoring.V2;
+using OmniSharp.Roslyn.CSharp.Services.Structure;
+using TestUtility;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace OmniSharp.MSBuild.Tests
+{
+    public class OnDemandProjectsLoadTests : AbstractMSBuildTestFixture
+    {
+        public OnDemandProjectsLoadTests(ITestOutputHelper output)
+            : base(output)
+        {
+        }
+
+        [Fact]
+        public async Task LoadProjectsOnDemandOneByOne()
+        {
+            var configData = new Dictionary<string, string> { [$"MsBuild:{nameof(MSBuildOptions.OnDemandProjectsLoad)}"] = "true" };
+            using (var testProject = await TestAssets.Instance.GetTestProjectAsync("TwoProjectsWithSolution"))
+            using (var host = CreateMSBuildTestHost(testProject.Directory, configurationData: configData))
+            {
+                MSBuildWorkspaceInfo workspaceInfo = await host.RequestMSBuildWorkspaceInfoAsync();
+
+                // Expect empty workspace initially since no documents have been requested yet
+                Assert.Null(workspaceInfo.SolutionPath);
+                Assert.NotNull(workspaceInfo.Projects);
+                Assert.Equal(0, workspaceInfo.Projects.Count);
+
+                // Requesting library document should load only that project
+                GetCodeActionsService codeActionHandler = host.GetRequestHandler<GetCodeActionsService>(OmniSharpEndpoints.V2.GetCodeActions);
+                GetCodeActionsResponse codeActionResponse = await codeActionHandler.Handle(
```

Comment:
I think both CodeActions and MembersAsTree use the GetDocuments flavor. Can you add a test that makes a request from the non-plural version?

Reply:
Actually Code Actions service does already request a single document: https://github.com/OmniSharp/omnisharp-roslyn/blob/24ab36e6e98beb21629f400e4905da53494e5a26/src/OmniSharp.Roslyn.CSharp/Services/Refactoring/V2/BaseCodeActionService.cs#L66